src/MyArrayList.java

```diff
@@ -0,0 +1,489 @@
+        // should be '20'and 190
+
+
+        System.out.println("Test #3.3 : MyArrayList#ListIterator.remove(Object)");
+        while (listIt.hasPrevious()) { //
+            Object el = listIt.previous();
+            if ((int)el%2==0) listIt.remove();
+        }
+        ((MyArrayList) lst).printAll();
+        System.out.println("");
+        // should be '1, 3, 5, 7..17, 19, null, ... null'
+
+
+        lst.clear();
+        System.out.println("Test #3.4 : MyArrayList#clear()");
+        System.out.println("Size is " + lst.size());
+        // should be 0
+
+
+        /**
+         * Finish
+         */
+        try {
+            System.out.println(lst.get(0));
+            System.out.println(lst.get(1));
+            throw new RuntimeException("Test didn't pass! List should be clean AF");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("All of Test successfully passed!");
+        } finally {
+        	System.out.println("\n\nEnd of the tests");
+        }
+
+    }
+
+    @Override
+    public Iterator<Object> iterator() { return new Itr(); }
+
+    private class Itr implements Iterator<Object> {
+        int cursor;       // index of next element to return
+        int lastRet = -1; // index of last element returned; -1 if no such
+
+        @Override
+        public boolean hasNext() {
+            return cursor != size;
+        }
+
+        @Override
+        public Object next() {
+            int i = cursor;
+            if (i >= size)
+                throw new NoSuchElementException();
+            cursor = i + 1;
+            return (Object) elements[lastRet = i];
+        }
+
+        @Override
+        public void remove() {
+            if (lastRet < 0)
+                throw new IllegalStateException();
+
+            try {
+                MyArrayList.this.remove(lastRet);
+                cursor = lastRet;
+                lastRet = -1;
+            } catch (IndexOutOfBoundsException ex) {
+                System.out.println(ex);
+            }
+        }
+    }
+
+    public ListIterator<Object> listIterator () { return new ListItr(0); }
+
+    private class ListItr extends Itr implements ListIterator<Object> {
+
+        public ListItr(int ind) {
+            super();
+            cursor = ind;
+        }
+
+        @Override
+        public boolean hasPrevious() {
+            return cursor != 0;
+        }
+
+        @Override
+        public Object previous() {
+            int i = cursor - 1;
+            if (i < 0)
+                throw new NoSuchElementException();
+            cursor = i;
+            return (Object) elements[lastRet = i];
+        }
+
+        @Override
+        public int nextIndex() {
+            return cursor;
+        }
+
+        @Override
+        public int previousIndex() {
+            return cursor - 1;
+        }
+
+        @Override
+        public void set(Object o) {
+            if (lastRet < 0)
+                throw new IllegalStateException();
```

Comment:
1) Need some explanation here. Not just an exception, but also informative message that explains for the user of the code what he did wrong.
2) Java Code Convention again. Sorry for that. But it is required. All if-else statements should contain curved brackets.